velox/exec/tests/HashJoinTest.cpp

```diff
@@ -31,212 +30,403 @@ using namespace facebook::velox::exec::test;
 
 using facebook::velox::test::BatchMaker;
 
+namespace {
 struct TestParam {
   int numDrivers;
 
   explicit TestParam(int _numDrivers) : numDrivers(_numDrivers) {}
 };
 
-class HashJoinTest : public HiveConnectorTestBase {
- protected:
-  HashJoinTest() : HashJoinTest(TestParam(1)) {}
+using SplitInput =
+    std::unordered_map<core::PlanNodeId, std::vector<exec::Split>>;
 
-  explicit HashJoinTest(const TestParam& param)
-      : numDrivers_(param.numDrivers), fuzzer_({}, pool_.get()) {}
+std::function<void(Task* task)> makeAddSplit(
+    bool& noMoreSplits,
+    SplitInput splits) {
+  return [&](Task* task) {
+    if (noMoreSplits) {
+      return;
+    }
+    for (auto& [nodeId, nodeSplits] : splits) {
+      for (auto& split : nodeSplits) {
+        task->addSplit(nodeId, std::move(split));
+      }
+      task->noMoreSplits(nodeId);
+    }
+    noMoreSplits = true;
+  };
+}
 
-  void SetUp() override {
-    HiveConnectorTestBase::SetUp();
+VectorFuzzer getFuzzer(
+    memory::MemoryPool* pool,
+    const VectorFuzzer::Options& options) {
+  return VectorFuzzer(options, pool);
+}
 
-    leftType_ =
-        ROW({{"t_k1", INTEGER()}, {"t_k2", VARCHAR()}, {"t_v1", VARCHAR()}});
-    rightType_ =
-        ROW({{"u_k1", INTEGER()}, {"u_k2", VARCHAR()}, {"u_v1", INTEGER()}});
+// Shuffle batches.
+void shuffleBatches(std::vector<RowVectorPtr>& batches) {
+  std::default_random_engine rng(1234);
+  std::shuffle(std::begin(batches), std::end(batches), rng);
+}
+
+// Make batches with random data.
+std::vector<RowVectorPtr> makeBatches(
+    int32_t batchSize,
+    int32_t numBatches,
+    RowTypePtr rowType,
+    memory::MemoryPool* pool,
+    double nullRatio = 0.1) {
+  std::vector<RowVectorPtr> batches;
+  batches.reserve(numBatches);
+  if (batchSize != 0) {
+    VectorFuzzer::Options options;
+    options.vectorSize = batchSize;
+    options.nullRatio = nullRatio;
+    VectorFuzzer fuzzer = getFuzzer(pool, options);
+    for (int32_t i = 0; i < numBatches; ++i) {
+      batches.push_back(fuzzer.fuzzRow(rowType));
+    }
+  } else {
+    batches.push_back(RowVector::createEmpty(rowType, pool));
+  }
+  // NOTE: we generate a number of vectors with a fresh new fuzzer init with
+  // the same fix seed. The purpose is to ensure we have sufficient match if
+  // we use the row type for both build and probe inputs. Here we shuffle the
+  // built vectors to introduce some randomness during the join execution.
+  shuffleBatches(batches);
+  return batches;
+}
+
+std::vector<RowVectorPtr> makeBatches(
```

Comment:
Any particular reason we are not shuffling batches here? This method looks similar to the one above, but doesn't shuffle. Is this intentional? Perhaps, document that behavior.